pkg/providers/github.go

```diff
@@ -0,0 +1,182 @@
+package providers
+
+import (
+	"context"
+	crypto_rand "crypto/rand"
+	"encoding/base64"
+	"errors"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/google/go-github/v43/github"
+	"github.com/spectralops/teller/pkg/core"
+	"golang.org/x/crypto/nacl/box"
+	"golang.org/x/oauth2"
+)
+
+const (
+	gitHubSplitPathCount = 2
+)
+
+// GitHubActionClient describe the GitHub action client
+type GitHubActionClient interface {
+	GetRepoPublicKey(ctx context.Context, owner, repo string) (*github.PublicKey, *github.Response, error)
+	CreateOrUpdateRepoSecret(ctx context.Context, owner, repo string, eSecret *github.EncryptedSecret) (*github.Response, error)
+	DeleteRepoSecret(ctx context.Context, owner, repo, name string) (*github.Response, error)
+	ListRepoSecrets(ctx context.Context, owner, repo string, opts *github.ListOptions) (*github.Secrets, *github.Response, error)
+}
+
+type GitHub struct {
+	clientActions GitHubActionClient
+}
+
+// NewGitHub create new GitHub provider
+func NewGitHub() (core.Provider, error) {
+
+	token := os.Getenv("GITHUB_AUTH_TOKEN")
```

Comment:
Is this what's default on CI system? I'm wondering if it's GITHUB_TOKEN. I know there's some shifting around it recently

Reply:
I used the same ENV var like they have in the [examples](https://github.com/google/go-github/tree/master/example).

You asking if `GITHUB_TOKEN` is better name right?

Reply:
tough call... ok lets leave it as is for now. thanks